<a href="https://colab.research.google.com/github/marce3-2140/Data-Analysis-with-Panda/blob/main/Titanic_predictions.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Ejercicios de Pandas en el conjunto de datos del Titanic

El objetivo de este dataset sacado de kaggle es predecir la supervivencia en el titanic a partir de las caracteristicas de los pasajeros. Para esto se responderan las siguientes preguntas.

**1. Muestren el centro del marco de datos.**

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd

In [3]:
filename = '/content/drive/MyDrive/CodingDojo/Fundamentals/Week2/Cargas/titanic.csv'
df = pd.read_csv(filename)

In [4]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


**2. ¿Cual es el porcentaje de las personas que sobrevivió?**

(pista: encuentren la media de la columna de supervivencia)

In [6]:
df_survived = df[df['Survived']==1]
df_survived['Survived'].value_counts()/len(df)*100

1    38.383838
Name: Survived, dtype: float64

*Respuesta:* Sobrevivio el 38.38% de los pasajeros

**3. ¿Cuántos de cada sexo sobrevivieron?**

In [7]:
df_survived['Sex'].value_counts()

female    233
male      109
Name: Sex, dtype: int64

*Respuesta:* Sobrevivieron 233 mujeres y 109 hombres.

**4. Si consideran solo el grupo de pasajeros que pago menos de $10, ¿Cuál sería el porcentaje de este grupo que sobrevivió?**

In [8]:
((df_survived['Fare']<10).sum())/len(df)*100

7.519640852974187

*Respuesta:* Considerando solo el grupo de sobrevivientes que pago menos de $10 su porcentaje de superviviencia fue de 7.5%

**5. En total, ¿Cuál es la edad promedio de los que no sobrevivieron?**

In [9]:
no_survived = df[df['Survived']==0]
no_survived['Age'].mean()

30.62617924528302

*Respuesta:* La edad promedio de los no sobrevivientes fue de 30 años.

**6. En total ¿Cuál es la edad promedio de los qe sobrevivieron?**

In [10]:
df_survived['Age'].mean()

28.343689655172415

*Respuesta:* La edad promedio de los sobrevientes fue de 28 años.

**7. ¿Cuál es el promedio de edad de los que sobrevivieron y los que no sobrevivieron agrupados por sexo?**

In [11]:
#Edad promedio de los que sobrevivieron agrupados por sexo
df_survived.groupby('Sex').mean()

<ipython-input-11-89584a92a540>:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df_survived.groupby('Sex').mean()


PassengerId  Survived    Pclass        Age     SibSp     Parch  \
Sex                                                                      
female   429.699571       1.0  1.918455  28.847716  0.515021  0.515021   
male     475.724771       1.0  2.018349  27.276022  0.385321  0.357798   

             Fare  
Sex                
female  51.938573  
male    40.821484

*Respuesta:* La edad promedio de los que sobrevivieron agrupados por sexo, observamos que fue de 28 años para mujeres y 27 años para hombres.

In [12]:
#Edad promedio de los que no sobrevivieron agrupados por sexo.
no_survived.groupby('Sex').mean()

<ipython-input-12-a15e2246c0f4>:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  no_survived.groupby('Sex').mean()


PassengerId  Survived    Pclass        Age     SibSp     Parch  \
Sex                                                                      
female   434.851852       0.0  2.851852  25.046875  1.209877  1.037037   
male     449.121795       0.0  2.476496  31.618056  0.440171  0.207265   

             Fare  
Sex                
female  23.024385  
male    21.960993

*Respuesta: * la edad promedio de los que no sobrevivieron agrupados por sexo fue para mujeres de 25 años y para los hombres fue de 31 años.

In [13]:
#Se muestran los dos datos para hombre y mujer agrupados con un groupby
df.groupby(['Survived','Sex'])['Age'].mean()

Survived  Sex   
0         female    25.046875
          male      31.618056
1         female    28.847716
          male      27.276022
Name: Age, dtype: float64